services/app-api/src/emailer/emails/resubmitPackageStateEmail.ts

```diff
@@ -16,20 +16,24 @@ import type { EmailData, EmailConfiguration } from '../'
 export const resubmitPackageStateEmail = async (
     pkg: LockedHealthPlanFormDataType,
     user: UserType,
-    resubmittedData: UpdatedEmailData,
-    config: EmailConfiguration
+    updateInfo: UpdateInfoType,
+    config: EmailConfiguration,
+    programs: ProgramType[]
 ): Promise<EmailData | Error> => {
     const isUnitTest = config.baseUrl === 'http://localhost'
     const isTestEnvironment = config.stage !== 'prod'
     const receiverEmails = generateStateReceiverEmails(pkg, user)
+    const packageName = generatePackageName(pkg, programs)
```

Comment:
I think we should move the potential error of not having a program that matches one of the programs we need into the emailer. Right now this call is safe because we checked that all the right programs exist in the resolver. I think it would be better to make that check closer to where we actually care about the programs existing, which is here in the emailer. You could do the same check here, returning an error if we are missing a program, and then handling that error in the resover the same way we handle any errors that the emailer could encounter.

It’s definitely a little six of one, half dozen the other, since we’re making the check either way, but it just trips me up a little that this generatePackageName call could fail to find the program it needs silently if we forget to do the check at the resolver level. It’s another case where I think we have sligtly less code as is, but at the cost of moving the error check outside the layer that actually cares about it.

Reply:
Done! I just moved `findPrograms` and `findAllPackageProgramIds` to the emailer functions. Returning error if `findProgams` encounters one.